flexget/plugins/sites/wordpress.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *  # noqa pylint: disable=unused-import, redefined-builtin
+from future.moves.urllib.parse import urlencode
+
+import logging
+
+import re
+from flexget import plugin
+from flexget.event import event
+from flexget.plugin import PluginWarning, PluginError
+from requests import Session
+from requests import Request, RequestException
+from requests.utils import dict_from_cookiejar, cookiejar_from_dict
+
+log = logging.getLogger('wordpress_auth')
+
+
+class WPLoginHeaders(dict):
```

Comment:
I'm not sure I like this. Why not just create a regular dict in `on_task_start`? Same goes for the other classes.

Reply:
I guess I'm pretty new to the python way of doing things. 
I created the WPLogin Classes so that I could have a very clean
```python
get_cookies(WPSession(), WPLoginRequest(url, username=username, password=password).prepare())
```
with WPSession and WPLoginRquest dependency injected into `get_cookies`
rather than having to imperatively write
```python
wpsession = Session()
wpsession.max_redirects = 5
wp_login_request = Request()
#etc
``` 
I felt it was a way to keep the `on_task_start` function short.
Again,
I don't have a very strong background in python, I'm more familiar with compiled languages.
So maybe this type of reasoning is faulty.